File: leetcode/C_01_array_two_pointers/D2023_11_14_Q283_move_zeroes/index_2.java
package leetcode.C_01_array_two_pointers.D2023_11_14_Q283_move_zeroes;

/**
 * 双指针解法
 */
public class index_2 {

    public static void main(String[] args) {
        System.out.println("index_2");
    }

    public void moveZeroes(int[] nums) {
        int i = 0;
        int j = 0;

        while (i < nums.length) {
            if (nums[i] != 0) {
                int tmp = nums[i];
                nums[i] = nums[j];
                nums[j] = tmp;

                j++;
            }

            i++;
        }

    }
}

File: leetcode/C_01_array_two_pointers/D2023_11_14_do_5_Q283_move_zeroes/index_2.ts
/**
 Do not return anything, modify nums in-place instead.
 */
// @ts-ignore
function moveZeroes(nums: number[]): void {
  let i: number = 0;
  let j: number = 0;

  while (j < nums.length) {
    if (nums[j] != 0) {
      const tmp = nums[j];
      nums[j] = nums[i];
      nums[i] = tmp;
      i++;
    }
    j++;
  }
}
